Lecture05

Vitald75 opened the pull request:
Use patterns: Page Object, Elements, Singleton, Builder, Chain of invocations, Data provider.
Ready methods: LoginPage, PimAddEmployeeTest (attempt using Auth Strategy via API or UI Login page).

Vitald75 committed:
Lecture05
First steps of refactor Page object

Vitald75 committed:
Lecture05
First steps of refactor Page object

VitaliDauhashei committed:
Page Object. Under construction

Vitald75 committed:
Lecture05
Refactor. Page object.

VitaliDauhashei committed:
Lecture05.
Refactor. Page Object.
Login, PIm

Luboval commented on src/main/java/eu/senla/core/Wait.java:
Возможно 7 лучше вынести в переменную

HannaYahaudzik commented on src/main/java/eu/senla/pages/DashBoardPage.java:
```diff
@@ -0,0 +1,39 @@
+  private String dashboardUrl =
+      "https://opensource-demo.orangehrmlive.com/web/index.php/dashboard/index";
+  private SidePanel sidePanel;
+  private final By sidePanelBody = By.cssSelector(".oxd-sidepanel-body");
```
Может стоит `sidePanelBody` перенести в класс `SidePanel`?

HannaYahaudzik commented on src/main/java/eu/senla/pages/LoginPage.java:
```diff
@@ -0,0 +1,68 @@
+    return this;
+  }
+
+  public final LoginPage loadPage(String url) {
```
Этот метод, наверное, лишний. Страница загружается в BaseTest

HannaYahaudzik commented on src/test/java/eu/senla/PimAddEmployeeTest.java:
```diff
@@ -1,47 +1,60 @@
-    driver.findElement(By.xpath("//button[@type='submit']")).click();
-
-    wait.until(d -> driver.findElement(By.xpath("//h6[text()='Alexandr PushyKin']"))).isDisplayed();
+    DashBoardPage dashBoardPage =
```
1. Можно вынести логику успешного прохождения формы Логин.
2. Значение `dashBoardPage` не используется после, значит можно объединить:
```
PIMPage pimPage =
        new LoginPage()
            .loadPage("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login")
            .loginAsValidUser("Admin", "admin123")
            .getSidePanel()
            .openPIMPage()
            .clickAddEmployeeButton()
            .fillNewEmployeeForm(employee)
            .clickSubmit()
            .isConfirmed()
            .isPersonalInformationPage();
```

HannaYahaudzik reviewed (approved):
Approve.
Комментарии к сведению.

DzianisShapel commented on src/main/java/eu/senla/data/Employee.java:
```diff
@@ -0,0 +1,16 @@
+@AllArgsConstructor
+@Builder
+@ToString
+@Getter
```
я вам не говорил, но есть такая аннотация `@Data`  https://projectlombok.org/features/Data 
можно одну ее поставить а 
@ToString
@Getter 
убрать и наверно 
@AllArgsConstructor тоже

DzianisShapel commented on src/test/java/eu/senla/LoginTest.java:
```diff
@@ -0,0 +1,52 @@
+
+    DashBoardPage loginPage =
+        new LoginPage()
+            .loadPage("https://opensource-demo.orangehrmlive.com/web/index.php/auth/login")
```
"https://opensource-demo.orangehrmlive.com/web/index.php/auth/login" убирай в проперти файл

DzianisShapel reviewed (changes requested):
в целом пожалуйста по удаляй весь закоменченный код, либо разкомментируй 
а то не совсем понятно, что валидно, а что нет 
на что смотреть, а что пропускать

VitaliDauhashei committed:
Lecture05.
Page Object.
Login, Logout, PIm tests.

refactor due to PR comments

Vitald75 committed:
Lecture05
Attemp to use strategy pattern, successfully extract cookie.

Vitald75 committed:
Merge remote-tracking branch 'origin/lecture05' into lecture05

# Conflicts:
#	src/main/java/eu/senla/pages/DashBoardPage.java
#	src/main/java/eu/senla/pages/PIMPage.java
#	src/test/java/eu/senla/BaseTest.java
#	src/test/java/eu/senla/PimAddEmployeeTest.java
#	src/test/java/eu/senla/resources/conf.properties

VitaliDauhashei committed:
Lecture05.
Refactor. Implement Authentication strategy.

Vitald75 committed:
Lecture05
Use Auth strategy to choose type of Login (via page or api), implement it in PimAddEmployeeTest.
Refactor.

Vitald75 committed:
Lecture05
Use Auth strategy to choose type of Login (via page or api), implement it in PimAddEmployeeTest.
Refactor.
Add RecruitmentCandidate page and test.

Commit:
start

Commit:
finished

VitaliDauhashei committed:
Lecture05.
Refactor. Implement Authentication strategy.
Move login strategy to the BaseTest
Split getCookie into two methods(steps)

VitaliDauhashei committed:
Lecture05.
Refactor. Implement Authentication strategy.
Move login strategy to the BaseTest
Split getCookie into two methods(steps)
